project-base/src/Shopsys/ShopBundle/Resources/scripts/styleguide/styleguide.js

```diff
@@ -0,0 +1,173 @@
+            });
+        } else {
+            this.$body.animate({ scrollTop: top - 50 }, 0);
+        }
+    },
+
+    /**
+   * Navigate to module on
+   * sidebar links click
+   */
+    navigateToAnchor: function ($elem) {
+
+        // Shutdown this feature in Chrome.
+        // Chrome have a know issue with file protocol and iframe comunication.
+        // It is not supported so we should not raise errors.
+        var top = this.$iframeContent.find('section' + $elem.attr('href')).offset().top - 50;
+
+        // Use ! to prevent de default browser behavior of anchor navigation
+        window.location.hash = '!' + $elem.attr('href').replace('#', '');
+
+        this.$body.animate({ scrollTop: top }, 800);
+    },
+    sidebarOpen: function () {
+        this.$body.addClass('opened');
+    },
+    sidebarClose: function () {
+        this.$body.removeClass('opened');
+    },
+
+    /**
+   * Leave or close the sidebar
+   * if the window is small
+   */
+    sidebarResizeHandler: function () {
+        if ($(window).width() <= 1220) {
+            this.sidebarClose();
+        } else {
+            this.sidebarOpen();
+        }
+    },
+    sidebarSetup: function () {
+
+        // Shutdown this feature in Chrome.
+        // Chrome have a know issue with file protocol and iframe comunication.
+        // It is not supported so we should not raise errors.
+
+        if ($(window).width() >= 1220) {
+            this.sidebarOpen();
+        }
+    },
+    setActiveSidebarLinkOnClick: function ($elem) {
+        this.$body.addClass(this.sidebarLinkWasClickedClass);
+        this.$sidebarLinks.removeClass(this.sidebarActiveLinkClass);
+        $elem.addClass(this.sidebarActiveLinkClass);
+    }
+};
+
+$(window).on('load', function () {
+    StyleguideIndex.init();
+
+    var textareas = document.querySelectorAll('.codemirror-html');
+    for (var i = 0; i < textareas.length; i++) {
+        CodeMirror.fromTextArea(textareas[i], {
+            lineNumbers: true,
+            theme: 'hopscotch',
+            mode: 'htmlmixed',
+            scrollbarStyle: 'simple',
+            extraKeys: { 'Ctrl-Space': 'autocomplete' }
+        });
+    }
+
+    // editor css
+    var textareas = document.querySelectorAll('.codemirror-css');
+    for (var i = 0; i < textareas.length; i++) {
+        CodeMirror.fromTextArea(textareas[i], {
+            lineNumbers: true,
+            theme: 'hopscotch',
+            mode: 'css',
+            scrollbarStyle: 'simple',
+            extraKeys: { 'Ctrl-Space': 'autocomplete' }
+        });
+    }
+
+    // editor js
+    var textareas = document.querySelectorAll('.codemirror-js');
+    for (var i = 0; i < textareas.length; i++) {
+        CodeMirror.fromTextArea(textareas[i], {
+            lineNumbers: true,
+            theme: 'hopscotch',
+            mode: 'javascript',
+            scrollbarStyle: 'simple',
+            extraKeys: { 'Ctrl-Space': 'autocomplete' }
+        });
+    }
+});
```

Comment:
This is the most beautiful part of javascript that I saw in fw

Reply:
Sad.. but maybe true :smile: